### PR TITLE
Add snowplow tracking for new records on TaskHistory

### PR DESCRIPTION
**Pull request by qnkhuat**
Product request: [thread](https://metaboat.slack.com/archives/C03E41P2E78/p1657215097262019).

If snowplow is enabled, we want to send every new `TaskHistory` to our snowplow cloud.
I implemented this as a `post-insert` hook to make sure all features TaskHistory is automatically recorded.

### CODE REVIEW

File: frontend/test/metabase/scenarios/onboarding/setup/setup.cy.spec.js [hunk 1]
```diff
@@ -242,28 +242,26 @@ describeWithSnowplow("scenarios > setup", () => {
   });
 
   it("should send snowplow events", () => {
-    // 1 - new_instance_created
-    // 2 - pageview
+    // 1 - pageview
```
Comment (qnkhuat) [inline frontend/test/metabase/scenarios/onboarding/setup/setup.cy.spec.js:245]: This is a tricky case.

Background on how we send the `new-instance-created` event:
- Whenever we start Metabase, we will initiate a SnowPlow instance,
- During the initiation of this instance, we check if `instance-creation` setting exits, if not we create one and send a `new_instance_created` event.
- After this, because `instance-creation` exists, we will not send `new-instance-created` event if users restart Metabase.

The reason this test is failing is that at the time of creating a snapshot for cypress test, the `instance-creation` already exists. So that is why even when we restore the database to `blank` state, we still don't see `new-instance-created` event being sent.

But why does `instance-creation` exists before we make a snapshot?
Because in this PR, we started to send snowplow events whenever we have a new task history, and that includes the sync, fingerprint, classify steps when we first start Metabase.
In short, the `new-instance-created` is sent even before the app is ready, thus `instance-creation` exists before we make a snapshot.

---

No worries, we still have the `new-instance-created` event being sent, it's just that we can't test it in cypress.
I hope it's okay? @nemanjaglumac

Comment (noahmoss) [reply frontend/test/metabase/scenarios/onboarding/setup/setup.cy.spec.js:245]: Seems fair enough to me.

In fact, I think setting `instance-creation` earlier is definitely an improvement. Currently, it's technically only set on the first page load, which theoretically could be some amount of time after the instance is actually started up. Now it'll be more accurate.

File: test/metabase/models/task_history_test.clj [hunk 2]
```diff
@@ -67,3 +71,62 @@
...
+  (db/insert! TaskHistory task)
+  (-> (snowplow-test/pop-event-data-and-user-id!)
+      last
+      mt/boolean-ids-and-timestamps
+      (update-in [:data "task_details"] json/parse-string)))
+
+(deftest snowplow-tracking-test
+  (snowplow-test/with-fake-snowplow-collector
+    (let [t1 (t/zoned-date-time)]
+      (testing "insert a task history should track a snowplow event"
```
Comment (noahmoss) [inline test/metabase/models/task_history_test.clj:87]: nit
```suggestion
      (testing "inserting a task history should track a snowplow event"
```

**Review by noahmoss — APPROVED** [review]
Cool, lgtm
